pkg/scheduler/apis/config/latest/latest_test.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package latest
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	componentbaseconfig "k8s.io/component-base/config"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/testing/defaults"
+)
+
+func TestGetInternalWithDefaults(t *testing.T) {
+
+	wantConfig := config.KubeSchedulerConfiguration{
+		Profiles: []config.KubeSchedulerProfile{
+			{
+				SchedulerName: "default-scheduler",
+				Plugins:       defaults.PluginsV1beta2,
+				PluginConfig:  defaults.PluginConfigs,
+			},
+		},
+		LeaderElection: componentbaseconfig.LeaderElectionConfiguration{
+			LeaderElect:       true,
+			LeaseDuration:     metav1.Duration{Duration: 15 * time.Second},
+			RenewDeadline:     metav1.Duration{Duration: 10 * time.Second},
+			RetryPeriod:       metav1.Duration{Duration: 2 * time.Second},
+			ResourceLock:      "leases",
+			ResourceName:      "kube-scheduler",
+			ResourceNamespace: "kube-system",
+		},
+		ClientConnection: componentbaseconfig.ClientConnectionConfiguration{
+			Kubeconfig:         "",
+			AcceptContentTypes: "",
+			ContentType:        "application/vnd.kubernetes.protobuf",
+			QPS:                50,
+			Burst:              100,
+		},
+		HealthzBindAddress: "0.0.0.0:10251",
+		MetricsBindAddress: "0.0.0.0:10251",
+		DebuggingConfiguration: componentbaseconfig.DebuggingConfiguration{
+			EnableProfiling:           true,
+			EnableContentionProfiling: true,
+		},
+		Parallelism:              16,
+		PodInitialBackoffSeconds: 1,
+		PodMaxBackoffSeconds:     10,
```

Comment:
Are these values (as well as QPS, Burst, etc.) defined somewhere as constants? If so, we should refer them here.

Reply:
Some of the constants are defined, some are not. For this case it doesn't matter because this is a test, it will fail if the value is changed.

Reply:
I'm honestly fine not having a test for this function :) We already have tests for defaulting. The rest is just calling one function.

Reply:
This is mostly to make sure it compiles.